src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/workflow/GerritReviewStep.java

```diff
@@ -0,0 +1,144 @@
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * Allows altering the Gerrit review posted at the end of build during the build.
+ * @author Teemu Murtola &lt;teemu.murtola@gmail.com&gt
+ */
+public class GerritReviewStep extends AbstractStepImpl {
+
+    private String customUrl;
+    private String unsuccessfulMessage;
+
+    /**
+     * Constructor.
+     *
+     * There are no mandatory parameters to the step.
+     */
+    @DataBoundConstructor
+    public GerritReviewStep() {
+    }
+
+    /**
+     * Gets the custom URL for a step.
+     * @return the URL.
+     */
+    @CheckForNull
+    public String getCustomUrl() {
+        return customUrl;
+    }
+
+    /**
+     * Sets a custom URL to post for a build.
+     * @param customUrl the URL to post.
+     */
+    @DataBoundSetter
+    public void setCustomUrl(String customUrl) {
+        this.customUrl = Util.fixEmptyAndTrim(customUrl);
+    }
+
+    /**
+     * Gets the unsuccessful message for a step.
+     * @return the message.
+     */
+    @CheckForNull
+    public String getUnsuccessfulMessage() {
+        return unsuccessfulMessage;
+    }
+
+    /**
+     * Sets additional information to post for a failed build.
+     * @param unsuccessfulMessage Additional information to post for a failed build.
+     */
+    @DataBoundSetter
+    public void setUnsuccessfulMessage(String unsuccessfulMessage) {
+        this.unsuccessfulMessage = Util.fixEmptyAndTrim(unsuccessfulMessage);
+    }
+
+    /**
+     * Executes the GerritReviewStep.
+     */
+    public static class Execution extends AbstractSynchronousStepExecution<Void> {
+
+        @StepContextParameter
+        private transient Run build;
+
+        @Inject
+        private transient GerritReviewStep step;
+
+        @Override
+        protected Void run() throws Exception {
+            ToGerritRunListener listener = ToGerritRunListener.getInstance();
+            String customUrl = step.getCustomUrl();
+            if (customUrl != null) {
+                listener.setBuildCustomUrl(build, customUrl);
+            }
+            String unsuccessfulMessage = step.getUnsuccessfulMessage();
+            if (unsuccessfulMessage != null) {
+                listener.setBuildFailureMessage(build, unsuccessfulMessage);
+            }
+            return null;
+        }
+
+        private static final long serialVersionUID = 1L;
+    }
+
+    /**
+     * Adds the step as a workflow extension.
+     */
+    @Extension(optional = true)
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        /**
+         * Constructor.
+         */
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "gerritReview";
```

Comment:
I don't like the name, mostly because it can imply that the review will be sent to Gerrit as the step is executed _(a feature I've been planning to implement for a while)_ It should have a name that implies properties being set rather than some immediate action. `setGerritReview` maybe?